iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/tsfile/TsFileResource.java

```diff
@@ -161,6 +161,9 @@ public class TsFileResource {
 
   private ProgressIndex maxProgressIndex;
 
+  /** used to prevent circular replication in PipeConsensus */
+  private boolean isGeneratedByPipeConsensus = false;
```

Comment:
do we need to serialize it?

Reply:
Seems no need. There is no serialization process involved from the generation of tsFileResource by PipeConsensusReceiver to the capture and process of the tsFileInsertionEvent by pipe

Reply:
> Seems no need. There is no deserialization process involved from the generation of tsFileResource by PipeConsensusReceiver to the capture and process of the tsFileInsertionEvent by pipe; For other processes involving deserialization of TsFileResource, TsFileResource will have a default value of false for `isGeneratedByPipeConsensus`, which does not affect correctness

Now we do the same thing to "isGeneratedByPipe" method and this is also a flag that will not recover. The consequence is that pipe may resend it after restarts, but seems that it does not affect the correctness(The complete correctness may be ensured after load tsfile is enriched with a progressIndex)

Reply:
@Caideyipi What you said makes sense, I will keep it in mind, thank you!